src/Compilers/CSharp/Portable/Compiler/MethodCompiler.cs

```diff
@@ -218,37 +218,51 @@ private void SetGlobalErrorIfTrue(bool arg)
                 }
             }
 
-            void addCircularStructDiagnostics(NamespaceOrTypeSymbol symbol)
+            void addAfterAccessorBindingDiagnostics(Symbol symbol)
             {
-                if (symbol is SourceMemberContainerTypeSymbol sourceMemberContainerTypeSymbol && PassesFilter(filterOpt, symbol))
+                if (PassesFilter(filterOpt, symbol))
                 {
-                    _ = sourceMemberContainerTypeSymbol.KnownCircularStruct;
+                    symbol.AfterAccessorBindingChecks();
                 }
 
-                foreach (var member in symbol.GetMembersUnordered())
+                if (symbol is not NamespaceOrTypeSymbol namespaceOrTypeSymbol)
                 {
-                    if (member is NamespaceOrTypeSymbol namespaceOrTypeSymbol)
+                    return;
+                }
+
+                foreach (var member in namespaceOrTypeSymbol.GetMembersUnordered())
+                {
+                    if (compilation.Options.ConcurrentBuild)
                     {
-                        if (compilation.Options.ConcurrentBuild)
-                        {
-                            Task worker = addCircularStructDiagnosticsAsAsync(namespaceOrTypeSymbol);
-                            methodCompiler._compilerTasks.Push(worker);
-                        }
-                        else
-                        {
-                            addCircularStructDiagnostics(namespaceOrTypeSymbol);
-                        }
+                        Task worker = addAfterAccessorBindingDiagnosticsAsAsync(member);
+                        methodCompiler._compilerTasks.Push(worker);
+                    }
+                    else
+                    {
+                        addAfterAccessorBindingDiagnostics(member);
+                    }
+
+                    var typeParameters = member switch
+                    {
+                        MethodSymbol m => m.TypeParameters,
+                        NamedTypeSymbol nt => nt.TypeParameters,
+                        _ => ImmutableArray<TypeParameterSymbol>.Empty,
+                    };
+
+                    foreach (var typeParameter in typeParameters)
+                    {
+                        (typeParameter as SourceTypeParameterSymbolBase)?.AfterAccessorBindingChecks();
                     }
```

Comment:
I'm not much confident about this.

Reply:
It looks like `PartialMethodWithArgumentConstraint` is now flaky. 😕 
@AlekseyTs @333fred Any suggestion on how to correctly avoid extra binding for type parameters constraints checks, and why the current approach is flaky?